File: src/main/java/ru/igrey/dev/dao/JdbcTemplateFactory.java
package ru.igrey.dev.dao;

import org.postgresql.jdbc2.optional.SimpleDataSource;
import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;

import static ru.igrey.dev.config.DealProps.*;

public class JdbcTemplateFactory {


    public NamedParameterJdbcTemplate projectx() {
        SimpleDataSource dataSource = new SimpleDataSource();
        dataSource.setUser(USER);
        dataSource.setPassword(PX_PASSWORD);
        dataSource.setServerName(PX_HOST);
        dataSource.setPortNumber(PX_PORT);
        dataSource.setDatabaseName(PX_DB);
        return new NamedParameterJdbcTemplate(dataSource);
    }

    public NamedParameterJdbcTemplate mastersber() {
        SimpleDataSource dataSource = new SimpleDataSource();
        dataSource.setUser(USER);
        dataSource.setPassword(MS_PASSWORD);
        dataSource.setServerName(HOST);
        dataSource.setPortNumber(MS_PORT);
        dataSource.setDatabaseName(MS_DB);
        return new NamedParameterJdbcTemplate(dataSource);
    }


}

File: src/main/kotlin/ru/igrey/dev/dao/JdbcTemplateFactory.kt
package ru.igrey.dev.dao

import org.postgresql.jdbc2.optional.SimpleDataSource
import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate
import ru.igrey.dev.config.DealProps.HOST
import ru.igrey.dev.config.DealProps.MS_DB
import ru.igrey.dev.config.DealProps.MS_DEAL_DB
import ru.igrey.dev.config.DealProps.MS_DEAL_HOST
import ru.igrey.dev.config.DealProps.MS_DEAL_PASSWORD
import ru.igrey.dev.config.DealProps.MS_DEAL_PORT
import ru.igrey.dev.config.DealProps.MS_PASSWORD
import ru.igrey.dev.config.DealProps.MS_PORT
import ru.igrey.dev.config.DealProps.PX_DB
import ru.igrey.dev.config.DealProps.PX_HOST
import ru.igrey.dev.config.DealProps.PX_PASSWORD
import ru.igrey.dev.config.DealProps.PX_PORT
import ru.igrey.dev.config.DealProps.USER

class JdbcTemplateFactory {

    fun projectx(): NamedParameterJdbcTemplate {
        val dataSource = SimpleDataSource()
        dataSource.user = USER
        dataSource.password = PX_PASSWORD
        dataSource.serverName = PX_HOST
        dataSource.portNumber = PX_PORT
        dataSource.databaseName = PX_DB
        return NamedParameterJdbcTemplate(dataSource)
    }

    fun mastersber(): NamedParameterJdbcTemplate {
        val dataSource = SimpleDataSource()
        dataSource.user = USER
        dataSource.password = MS_PASSWORD
        dataSource.serverName = HOST
        dataSource.portNumber = MS_PORT
        dataSource.databaseName = MS_DB
        return NamedParameterJdbcTemplate(dataSource)
    }

    fun jdbcTemplateMSDeal(): NamedParameterJdbcTemplate {
        val dataSource = SimpleDataSource()
        dataSource.user = USER
        dataSource.password = MS_DEAL_PASSWORD
        dataSource.serverName = MS_DEAL_HOST
        dataSource.portNumber = MS_DEAL_PORT
        dataSource.databaseName = MS_DEAL_DB
        return NamedParameterJdbcTemplate(dataSource)
    }
}
